File: righttriangle/righttriangle.go
package main

import "fmt"

type Answer struct {
	m int
	n int
	p int
}

func solve() []Answer {
	answers := []Answer{}
	for m := 1; m < 30; m++ {
		for n := 1; n < 30; n++ {
			p2 := m*m + n*n
			ok, p := isSquareNumber(p2)
			if ok {
				answers = append(answers, Answer{m, n, p})
			}
		}
	}
	return answers
}

func isSquareNumber(p2 int) (bool, int) {
	for i := 1; i < p2; i++ {
		if p2 == i*i {
			return true, i
		}
	}
	return false, 0
}

func main() {
	answers := solve()
	for _, ans := range answers {
		fmt.Printf("%v\n", ans)
	}
}

File: src/right_triangle.rs
#[derive(Debug)]
struct Answer {
    m: u32,
    n: u32,
    p: u32,
}

impl Answer {
    fn new(m:u32, n:u32, p:u32) -> Answer {
        Answer{m, n, p}
    }
}

fn solve() -> Vec<Answer>{
    let mut answers = Vec::new();
    for m in 1..30 {
        for n in 1..30 {
            let p2 = m * m + n * n;
            if let Some(p) = is_square_number(p2) {
                let ans = Answer::new(m, n, p);
                answers.push(ans);
            }
        }
    }
    answers
}

fn is_square_number(n:u32) -> Option<u32> {
    for i in 1..n {
        if n == i * i {
            return Some(i)
        }
    }
    return None;
}

#[cfg(test)]
mod tests {
    #[test]
    fn test_solve(){
        let answers = super::solve();
        println!("{:?}", answers);
    }
}